### PR TITLE
[AVS] hide sections without data

### PR DESCRIPTION
**Pull request by acrollet**
## Summary

- hide "New Orders from this appointment" if there is no content below it
- empty individual vitals should be hidden, whole section should be hidden if there are only empty records

## Testing done

- local testing
- added unit tests

## What areas of the site does it impact?

AVS application

## Acceptance criteria

- [ ] New orders heading should only be shown if there is orders content
- [ ] Vitals section should only be shown if there is non-empty vitals content

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### CODE REVIEW

**Review by stephenBarrs — APPROVED** [review]
LGTM

File: src/applications/avs/components/ItemsBlock.jsx [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
 const ItemsBlock = props => {
   const { heading, intro, itemType, items, renderItem, showSeparators } = props;
 
-  if (!items || items.length === 0) {
-    return null;
-  }
+  if (!items) return null;
+
+  // Filter out null/empty field values.
+  let listItems = items.filter(item => !allFieldsEmpty(item));
```
Comment (mdewey) [inline src/applications/avs/components/ItemsBlock.jsx:12]: NITPICK: Is there any chance that items will not be an array?

**Review by mdewey — APPROVED** [review]
LGTM!

File: src/applications/avs/components/ItemsBlock.jsx (hunk 1 above)
Comment (acrollet) [reply src/applications/avs/components/ItemsBlock.jsx:12]: Items _shouldn't_ be anything other than an array, but it's a good point. I think what I'll do is make a separate PR to update ItemsBlock and the other formatting components to throw an error if they encounter a wrong type. (On the theory that it's better to bomb out than show something incorrect or not show data that should be there)

**Pull request merged**